### PR TITLE
Allow user to specify registry credentials in values or via a premade secret.

### PR DESCRIPTION
**Pull request by paulczar**

### CODE REVIEW

File: stable/spinnaker/templates/configmap/halyard-config.yaml [hunk 1]
```diff
@@ -63,8 +66,18 @@ data:
...
+    {{ if $registry.email -}}
+    CREDS+=" --email {{ $registry.email }}"
+    {{- end -}}
+    {{- end }}
+
     $HAL_COMMAND config provider docker-registry account $PROVIDER_COMMAND {{ $registry.name }} --address {{ $registry.address }} \
-        --repositories {{ range $index, $repository := $registry.repositories }}{{- $repository }},{{- end }}
+        ${CREDS} --repositories {{ range $index, $repository := $registry.repositories }}{{if $index}},{{end}}{{- $repository }}{{- end }}
+
+    sleep 600
```
Comment (viglesiasce) [inline stable/spinnaker/templates/configmap/halyard-config.yaml]: Is the sleep needed?

File: stable/spinnaker/values.yaml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 halyard:
+  debugConfig: true
```
Comment (viglesiasce) [inline stable/spinnaker/values.yaml]: What does this do?

File: stable/spinnaker/templates/configmap/halyard-config.yaml [hunk 3]
```diff
@@ -54,6 +56,7 @@ data:
     {{ end }}
 
     # Docker Registry
+
```
Comment (viglesiasce) [inline stable/spinnaker/templates/configmap/halyard-config.yaml]: remove whitespace

File: stable/spinnaker/values.yaml [hunk 4]
```diff
@@ -31,6 +32,15 @@ dockerRegistries:
 #   password: '<INSERT YOUR SERVICE ACCOUNT JSON HERE>'
 #   email: 1234@5678.com
 
+# If you don't want to put your passwords into a values file
```
Comment (viglesiasce) [inline stable/spinnaker/values.yaml:34]: Can you add something about this to the README as well?

File: stable/spinnaker/templates/statefulsets/halyard.yaml [hunk 5]
```diff
@@ -68,6 +68,13 @@ spec:
         secret:
           secretName: {{ template "spinnaker.fullname" . }}-gcs
       {{- end }}
+      - name: reg-secrets
+        secret:
+          {{- if .Values.accountSecret }}
```
Comment (viglesiasce) [inline stable/spinnaker/templates/statefulsets/halyard.yaml]: I think this should be `if .Values.dockerRegistryAccountSecret`

**Review by viglesiasce — CHANGES_REQUESTED** [review]

File: stable/spinnaker/values.yaml (hunk 4 above)
Comment (paulczar) [reply stable/spinnaker/values.yaml:34]: done!

File: stable/spinnaker/templates/configmap/halyard-config.yaml (hunk 3 above)
Comment (paulczar) [reply stable/spinnaker/templates/configmap/halyard-config.yaml]: done

File: stable/spinnaker/templates/configmap/halyard-config.yaml (hunk 1 above)
Comment (paulczar) [reply stable/spinnaker/templates/configmap/halyard-config.yaml]: no, accidently left in from debugging.

**Comment by viglesiasce** [comment]
Thanks @paulczar!

**Review by viglesiasce — APPROVED** [review]

**Pull request merged**